moveit_core/utils/include/moveit/utils/robot_model_builder.h

```diff
@@ -0,0 +1,178 @@
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF AD w/ 2VISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Bryce Willey */
+
+#ifndef MOVEIT_CORE_UTILS_TEST_
+#define MOVEIT_CORE_UTILS_TEST_
+
+/** \file test_utils.h
+ *  \brief convenience functions and classes used for making simple robot models for testing.
+ */
+#include <urdf_parser/urdf_parser.h>
+#include <urdf_model/model.h>
+#include <urdf_model/joint.h>
+#include <srdfdom/srdf_writer.h>
+#include <urdf/model.h>
+#include <fstream>
+#include <boost/filesystem/path.hpp>
+#include <moveit_resources/config.h>
+#include <moveit/robot_model/robot_model.h>
+#include <geometry_msgs/Pose.h>
+
+namespace moveit
+{
+namespace core
+{
+/** \brief Loads a robot from moveit_resources.
+ * \param[in] robot_name The name of the robot package in moveit_resources to load.
+ *            For example, "panda_description", or "faunc_description".
+ * \returns a RobotModel constructed from robot_name's URDF and SRDF.
+ */
+moveit::core::RobotModelPtr loadRobot(std::string robot_name);
+
+urdf::ModelInterfaceSharedPtr loadURDF(std::string robot_name);
+
+srdf::ModelSharedPtr loadSRDF(std::string robot_name);
+
+/** \brief Easily build different robot models for testing.
+ *  Essentially a programmer-friendly light wrapper around URDF and SRDF.
+ *  Best shown by an example:
+ *  \code{.cpp}
+ *  RobotModelBuilder builder("my_robot", "base_link");
+ *  builder.add("a->b->c", "continuous");
+ *  builder.addGroup({"a", "b"}, {}, "example_group");
+ *  RobotModelPtr model = builder.build();
+ *  \endcode
+ */
+class RobotModelBuilder
+{
+public:
+  /** \brief Constructor, takes the names of the robot and the base link.
+   * \param[in] name The name of the robot, i.e. the 'name' attribute of the robot tag in URDF
+   * \param[in] base_link_name The name of the base/root link of the robot. All other links should be descendants
+   *            of this link
+   */
+  RobotModelBuilder(std::string name, std::string base_link_name);
+
+  /** \name URDF Functions
+      \{ */
+
+  /** \brief Adds a chain of links and joints to the builder.
+   *  The joint names are generated automatically as "<parent>-<child>-joint".
+   * \param[in] section A list of link names separated by "->". The first link should already be added to the build by
+   * the time this function is called
+   * \param[in] type The type of the joints connecting all of the given links, e.g. "revolute" or "continuous". All of
+   * the joints will be given this type. To add multiple types of joints, call this method multiple times
+   * \param[in] joint_origins The "parent to joint" origins for the joints connecting the links. If not used, all
+   * origins will default to the identity transform
+   */
+  void add(std::string section, std::string type, std::vector<geometry_msgs::Pose> joint_origins = {});
```

Comment:
```suggestion
  void addChain(std::string section, std::string type, std::vector<geometry_msgs::Pose> joint_origins = {});
```
more descriptive function names